intel-retail/automated-self-checkout
# feat: Each OVMS server has its own config json

Make each ovms server have its own unique config json instead of sharing the same config.json for all instances of ovms servers

- removed the old way of updating devices and artifacts
- updating device logic is now in profile-launcher
- the config json file name is derived from each ovms server docker instance container name per instance
- clean up all config json files on make clean-all or make clean-ovms-server

  FIXES: #322

## PR Checklist



## What are you changing?


## Issue this PR will close

close: #**322**

## Anything the reviewer should know when reviewing this PR?
- based on the design doc: https://github.com/intel-retail/documentation/pull/13  

## Test Instructions if applicable

- git clone this PR
- unit tests: run `make unit-test-profile-launcher`  should see ALL passed
- functional test: run `./test_device_switch.sh`  this is the script to mimic the situation shown in issue #322 running two instances of pipelines and ovms servers and switching from CPU to GPU.0.  After you ran this script you should see both pipelines are running OK and displayed now and also should see PASSED on the test script itself.
Example output on the terminal:
```console
PASSS: OVMS server config_ovms-server0.json exists for the instance of Docker ovms-server0
PASSS: OVMS server config_ovms-server1.json exists for the instance of Docker ovms-server1
[2023-11-16 21:38:34.197][1][modelmanager][info][modelmanager.cpp:121] Available devices for Open VINO: CPU, GNA
[2023-11-16 21:38:34.199][1][serving][info][modelinstance.cpp:910] Loading model: bit_64, version: 1, from path: /models/BiT_M_R50x1_10C_50e_IR/FP16-INT8/1, with target device: CPU ...
[2023-11-16 21:38:34.732][1][modelmanager][info][modelinstance.cpp:715] Plugin config for device: CPU
[2023-11-16 21:38:34.742][1][serving][info][modelinstance.cpp:910] Loading model: instance-segmentation-security-1040, version: 1, from path: /models/instance-segmentation-security-1040/FP16-INT8/1, with target device: CPU ...
[2023-11-16 21:38:35.681][1][modelmanager][info][modelinstance.cpp:715] Plugin config for device: CPU
[2023-11-16 21:38:35.695][1][serving][info][modelinstance.cpp:910] Loading model: yolov5s, version: 1, from path: /models/yolov5s/FP16-INT8/1, with target device: CPU ...
[2023-11-16 21:38:38.293][1][modelmanager][info][modelinstance.cpp:715] Plugin config for device: CPU
[2023-11-16 21:38:38.297][1][serving][info][modelinstance.cpp:910] Loading model: efficientnet-b0, version: 1, from path: /models/efficientnet-b0/FP32/1, with target device: CPU ...
[2023-11-16 21:38:38.415][1][modelmanager][info][modelinstance.cpp:715] Plugin config for device: CPU
[2023-11-16 21:38:38.419][1][serving][info][modelinstance.cpp:910] Loading model: ssd_mobilenet_v1_coco, version: 1, from path: /models/ssd_mobilenet_v1_coco/FP32/1, with target device: CPU ...
[2023-11-16 21:38:38.499][1][modelmanager][info][modelinstance.cpp:715] Plugin config for device: CPU
[2023-11-16 21:38:38.502][1][serving][info][modelinstance.cpp:910] Loading model: face-detection-retail-0005, version: 1, from path: /models/face-detection-retail-0005/FP16-INT8/1, with target device: CPU ...
[2023-11-16 21:38:38.717][1][modelmanager][info][modelinstance.cpp:715] Plugin config for device: CPU
[2023-11-16 21:38:56.547][1][modelmanager][info][modelmanager.cpp:121] Available devices for Open VINO: CPU, GNA, GPU
[2023-11-16 21:38:56.549][1][serving][info][modelinstance.cpp:910] Loading model: bit_64, version: 1, from path: /models/BiT_M_R50x1_10C_50e_IR/FP16-INT8/1, with target device: GPU.0 ...
[2023-11-16 21:38:57.527][1][modelmanager][info][modelinstance.cpp:715] Plugin config for device: GPU.0
[2023-11-16 21:38:57.531][1][serving][info][modelinstance.cpp:910] Loading model: instance-segmentation-security-1040, version: 1, from path: /models/instance-segmentation-security-1040/FP16-INT8/1, with target device: GPU.0 ...
[2023-11-16 21:38:59.001][1][modelmanager][info][modelinstance.cpp:715] Plugin config for device: GPU.0
[2023-11-16 21:38:59.011][1][serving][info][modelinstance.cpp:910] Loading model: yolov5s, version: 1, from path: /models/yolov5s/FP16-INT8/1, with target device: GPU.0 ...
[2023-11-16 21:39:06.301][1][modelmanager][info][modelinstance.cpp:715] Plugin config for device: GPU.0
[2023-11-16 21:39:06.306][1][serving][info][modelinstance.cpp:910] Loading model: efficientnet-b0, version: 1, from path: /models/efficientnet-b0/FP32/1, with target device: GPU.0 ...
[2023-11-16 21:39:06.496][1][modelmanager][info][modelinstance.cpp:715] Plugin config for device: GPU.0
[2023-11-16 21:39:06.498][1][serving][info][modelinstance.cpp:910] Loading model: ssd_mobilenet_v1_coco, version: 1, from path: /models/ssd_mobilenet_v1_coco/FP32/1, with target device: GPU.0 ...
[2023-11-16 21:39:06.635][1][modelmanager][info][modelinstance.cpp:715] Plugin config for device: GPU.0
[2023-11-16 21:39:06.636][1][serving][info][modelinstance.cpp:910] Loading model: face-detection-retail-0005, version: 1, from path: /models/face-detection-retail-0005/FP16-INT8/1, with target device: GPU.0 ...
[2023-11-16 21:39:06.996][1][modelmanager][info][modelinstance.cpp:715] Plugin config for device: GPU.0
```
- you can also run smoke-test if you want and you should see ALL PASSED

## If the there are associated PRs in other repositories, please link them here (i.e. intel-retail/automated-self-checkout )
